api/client.go

```diff
@@ -258,8 +258,8 @@ func (c Client) CreateUser(ctx context.Context, req *CreateUserRequest) (*Create
 	return post[CreateUserResponse](ctx, c, "/api/users", req)
 }
 
-func (c Client) UpdateUser(ctx context.Context, req *UpdateUserRequest) (*User, error) {
-	return put[User](ctx, c, fmt.Sprintf("/api/users/%s", req.ID.String()), req)
+func (c Client) UpdateUser(ctx context.Context, req *UpdateUserRequest) (*UpdateUserResponse, error) {
```

Comment:
Shouldn't this stay `User`? As a guideline for our API we return the Resource for `PUT` requests: https://github.com/infrahq/infra/blob/main/docs/api/guidelines.md#method-calls

Reply:
As a general rule, that's true but there have been exceptions. For example, `POST` requests are also supposed to return the resource but `POST /api/users` returns a different response since it needs to return the `OneTimePassword`. Same thing with `POST /api/access-keys` which needs to return the secret. 

In order to have the server generate the new password on reset, this change is needed. There are two alternatives if following these guidelines is required:

1. Add a separate endpoint strictly for an admin to reset passwords
2. Change `api.User` to always include `OneTimePassword` but leave it unset in non-CreateUser/-UpdateUser cases and have both `POST /api/users` and `PUT /api/users/:d` return `api.User`